ui/app/components/secret-delete-menu.js

```diff
@@ -0,0 +1,147 @@
+import Ember from 'ember';
+import { inject as service } from '@ember/service';
+import Component from '@glimmer/component';
+import { tracked } from '@glimmer/tracking';
+import { action } from '@ember/object';
+import { alias } from '@ember/object/computed';
+import { maybeQueryRecord } from 'vault/macros/maybe-query-record';
+
+const getErrorMessage = errors => {
+  let errorMessage = errors?.join('. ') || 'Something went wrong. Check the Vault logs for more information.';
+  return errorMessage;
+};
+export default class SecretDeleteMenu extends Component {
+  @service store;
+  @service router;
+  @service flashMessages;
+
+  @tracked showDeleteModal = false;
+
+  @maybeQueryRecord(
+    'capabilities',
+    context => {
+      if (!context.args.model) {
+        return;
+      }
+      let backend = context.args.model.backend;
+      let id = context.args.model.id;
+      let path = context.args.isV2 ? `${backend}/data/${id}` : `${backend}/${id}`;
```

Comment:
Great point.  Let me check this out.

Reply:
Note to self: it errors if unsafe id or backend before this point.  Something to revisit.

Reply:
I think it's worth urlEncoding here even though it currently fails upstream, so that in the future if/when the upstream thing is fixed we don't have unsafe strings here